shell/platform/embedder/embedder.cc

```diff
@@ -2998,11 +2999,13 @@ FlutterEngineResult FlutterEngineNotifyDisplayUpdate(
     case kFlutterEngineDisplaysUpdateTypeStartup: {
       std::vector<std::unique_ptr<flutter::Display>> displays;
       for (size_t i = 0; i < display_count; i++) {
+        const auto* display = &embedder_displays[i];
```

Comment:
To fix https://github.com/flutter/flutter/issues/121352 you'll need to do pointer math instead of array indexing. This is necessary as the embedder might not have all the members on the `FlutterEngineDisplay` struct.

For example, see how the pointer events are "indexed":

https://github.com/flutter/engine/blob/6dc9b28677b7d3e65594864347a2c1e3c1b19481/shell/platform/embedder/embedder.cc#L2250-L2251

Reply:
Done

Reply:
Looks good, thank you for fixing this!